Update gnomAD Hail Tables Help page (#1430)
broadinstitute/gnomad-browser: browser/help/topics/v4-hts.md

  - `freq_meta`: Allele frequency metadata for the gnomAD exomes. An ordered list containing the frequency aggregation group for each element of the `exomes.freq` array row annotation.
  - `freq_meta_sample_count`: A sample count per sample grouping defined in the exomes `exomes.freq_meta` global annotation.
  - `faf_meta`: Filtering allele frequency metadata for the gnomAD exomes. An ordered list containing the frequency aggregation group for each element of the `exomes.faf` array row annotation.
  - `age_distribution`: Callset-wide age histogram calculated on the gnomAD exomes.
    - `bin_edges`: Bin edges for the age histogram.
    - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
    - `n_smaller`: Count of age values falling below lowest histogram bin edge.
    - `n_larger`: Count of age values falling above highest histogram bin edge.
- `joint_globals`: Global fields from the combined (joint) gnomAD exomes and genomes.
  - `freq_meta`: Allele frequency metadata for the joint gnomAD exomes and genomes. An ordered list containing the frequency aggregation group for each element of the `joint.freq` array row annotation.

>   - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the `joint.freq` array row annotation.

Reviewer: @jkgoodrich `exomes_globals` and `genomes_globals` don't have `freq_index_dict` nor `faf_index_dict`, do we need to add them in?

Author: resolving this now that you've added them

  - `fail_interval_qc`
  - `outside_ukb_capture_region`
  - `outside_broad_capture_region`
- `allele_info`: The v4 exomes Hail Table has an additional field in this struct, 'has_star', that is not present in the struct on the v4 genomes Hail Table.
- `info`: Sibling singletons were used to train the variant QC models for the v4 exomes but were not used in the v4 genomes variant QC.

#### gnomAD v4.1 joint frequency Hail Table annotations

The v4.1 joint (combined exomes + genomes) frequency Hail Table only contains frequencies for the following groupings:
- `group`
- `sex` (`adj`<sup>1</sup> only)
- `gen_anc` (`adj`<sup>1</sup> only)
- `gen_anc` and `sex` (`adj`<sup>1</sup> only)

1. Includes only genotypes with depth >= 10, genotype quality >= 20 and minor allele balance > 0.2 for heterozygous genotypes.

Global fields

- `versions`: Struct containing the exomes and genomes versions.
- `genomes_globals`: Global fields from the gnomAD genomes.
  - `freq_meta`: Allele frequency metadata for the gnomAD genomes. An ordered list containing the frequency aggregation group for each element of the `genomes.freq` array row annotation.

>   - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the `genomes.freq` array row annotation.

Author: ```suggestion
  - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group [adj only], sex: sex karyotype [adj only]), with values describing the corresponding index of each grouping entry in the `genomes.freq` array row annotation.
```

  - `faf_meta`: Filtering allele frequency metadata for the gnomAD genomes. An ordered list containing the frequency aggregation group for each element of the `genomes.faf` array row annotation.
  - `faf_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (`genomes.faf`) row annotation.
  - `age_distribution`: Callset-wide age histogram calculated on the gnomAD genomes.
    - `bin_edges`: Bin edges for the age histogram.
    - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
    - `n_smaller`: Count of age values falling below lowest histogram bin edge.
    - `n_larger`: Count of age values falling above highest histogram bin edge.
- `exomes_globals`: Global fields from the gnomAD exomes.
  - `freq_meta`: Allele frequency metadata for the gnomAD exomes. An ordered list containing the frequency aggregation group for each element of the `exomes.freq` array row annotation.

>   - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the `exomes.freq` array row annotation. 

Author: ```suggestion
  - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group [adj only], sex: sex karyotype [adj only]), with values describing the corresponding index of each grouping entry in the `exomes.freq` array row annotation. 
```

  - `freq_meta_sample_count`: A sample count per sample grouping defined in the genomes `genomes.freq_meta` global annotation.
  - `faf_meta`: Filtering allele frequency metadata for the gnomAD genomes. An ordered list containing the frequency aggregation group for each element of the `genomes.faf` array row annotation.
  - `faf_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (`genomes.faf`) row annotation.
  - `age_distribution`: Callset-wide age histogram calculated on the gnomAD genomes.
    - `bin_edges`: Bin edges for the age histogram.
    - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
    - `n_smaller`: Count of age values falling below lowest histogram bin edge.
    - `n_larger`: Count of age values falling above highest histogram bin edge.

> - `joint_globals`: Global fields from the combined (joint) gnomAD exomes and genomes.

Author: should we add a couple extra notes here reminding users that `ami` contains only genome samples (in `freq`) and `mid` contains only exome counts (for `faf`)?

  - `freq_meta`: Allele frequency metadata for the joint gnomAD exomes and genomes. An ordered list containing the frequency aggregation group for each element of the `joint.freq` array row annotation.

>   - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the `joint.freq` array row annotation.

Author: ```suggestion
  - `freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group [adj only], sex: sex karyotype [adj only]), with values describing the corresponding index of each grouping entry in the `joint.freq` array row annotation.
```

#### Version 4.1 sample grouping combinations and 'freq' array access

The available v4.1 grouping combinations within the 'freq' array annotation are listed below. adj<sup>1</sup> must be provided as the “group” for all combinations except when requesting raw frequency information, which is only available for the main gnomAD callsets and subsets.

- group, e.g. “`adj`”, “`raw`”
- sex-group, e.g. “`XX_adj`”
- subset-group, e.g. “`non_ukb-raw`”
- gen-anc<sup>3</sup>-group, e.g. “`afr_adj`”
- gen-anc-sex-group, e.g. “`ami_XX_adj`”
- downsampling<sup>2</sup>-group-gen-anc, e.g. “`10_adj_eas`”,
- subset-gen-anc-group, e.g. “`_sas_adj`”
- subset-sex-group, e.g. “`non_ukb_XY_adj`”
- subset-gen-anc<sup>3</sup>-sex-group, e.g. “`non_ukb_mid_XX_adj`”,

To access the '`freq`' array using the '`freq_index_dict`', you need to retrieve the value of your desired label combination key. The example below accesses the entry of the high quality genotypes (group: adj) of XX individuals (sex: XX) clustered with the AFR genetic ancestry group in the gnomAD v4.1 exomes:

```
# Load the v4.1 exomes public release HT
from gnomad.resources.grch38.gnomad import public_release

> ht = public_release(“exomes”).ht()

Reviewer: ```suggestion
ht = public_release("exomes").ht()
```

| ---------------------------- | -------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `group`                      | Genotype's filter                      | adj<sup>1</sup>, raw                                                                                                                                                                                                                                                                           | adj<sup>1</sup>, raw                                                                                                                                                                                                                                                                                                                                                                                                                              |

Author: good catch! that must have been from from when we had this as the second point:
> The labels we use to classify individuals by chromosomal sex changed from “male” and “female” to “XY” and “XX.” More details available in this [blog post](https://gnomad.broadinstitute.org/news/2020-10-gnomad-v3-1-new-content-methods-annotations-and-data-availability/#tweaks-and-updates).

will remove

| `gen_anc`                    | gnomAD inferred genetic ancestry group | `afr`, `amr`, `asj`, `eas`, `fin`, `mid`, `nfe`, `rmi`, `sas`                                                                                                                                                                                                                                  | `afr`, `amr`, `asj`, `eas`, `fin`, `mid`, `nfe`, `rmi`, `sas`                                                                                                                                                                                                                                                                                                                                                                                     |

> | `gen_anc` (1KG subset only)<sup>3</sup>  | The 1KG project's ancestry             | N/A                                                                                                                                                                                                                                                                                            | `acb`, `asw`, `beb`, `cdx`, `ceu`, `chb`, `chs`, `clm`, `esn`, `fin`, `gbr`, `gih`, `gwd`, `ibs`, `itu`, `jpt`, `khv`, `lwk`, `msl`, `mxl`, `pel`, `pjl`, `pur`, `stu`, `tsi`, `yri`                                                                                                                                                                                                                                                              |

Author: I think this is also leftover from when we had a different second bullet (https://github.com/broadinstitute/gnomad-browser/pull/1430#discussion_r1537682175)

#### gnomAD v4.1 genomes Hail Table annotations

The v4 genomes Hail Table annotation schema is the same as the exomes schema, with only a few minor differences:

Global fields

- `interval_qc_parameters`: Only the v4 exomes HT has this global annotation

Row fields

- `fafmax`: This annotation is stratified by subset in the v4 exomes, but the v4 genomes annotation does not have any subset stratification
- `region_flags`: The v4 exomes Hail Table has the following fields that are not present in the struct on the v4 genomes Hail Table:
  - `fail_interval_qc`
  - `outside_ukb_capture_region`
  - `outside_broad_capture_region`
- `allele_info`: The v4 exomes Hail Table has an additional field in this struct, 'has_star', that is not present in the struct on the v4 genomes Hail Table
- `info`: Sibling singletons were used to train the variant QC models for the v4 exomes but were not used in the v4 genomes variant QC

#### gnomAD v4.1 joint frequency Hail Table annotations

> The v4 genomes Hail Table annotation schema is the same as the exomes schema, with only a few minor differences:

Reviewer: This description has to be changed.

> Global fields

Reviewer: The Global fields have changed as in: : gs://gnomad-tmp/release/4.1/ht/joint/gnomad.joint.v4.1.faf.filtered.ht
```
Global fields:
    'versions': struct {
        exomes: str, 
        genomes: str
    } 
    'genomes_globals': struct {
        freq_meta: array<dict<str, str>>, 
        freq_meta_sample_count: array<int32>, 
        faf_meta: array<dict<str, str>>, 
        age_distribution: struct {
            bin_edges: array<float64>, 
            bin_freq: array<int32>, 
            n_smaller: int32, 
            n_larger: int32
        }
    } 
    'exomes_globals': struct {
        freq_meta: array<dict<str, str>>, 
        freq_meta_sample_count: array<int32>, 
        faf_meta: array<dict<str, str>>, 
        age_distribution: struct {
            bin_edges: array<float64>, 
            bin_freq: array<int32>, 
            n_smaller: int32, 
            n_larger: int32
        }
    } 
    'joint_globals': struct {
        freq_meta: array<dict<str, str>>, 
        freq_index_dict: dict<str, int32>, 
        faf_meta: array<dict<str, str>>, 
        faf_index_dict: dict<str, int32>, 
        freq_meta_sample_count: array<int32>, 
        age_distribution: struct {
            bin_edges: array<float64>, 
            bin_freq: array<int32>, 
            n_smaller: int32, 
            n_larger: int32
        }
    } 
```

- `joint_freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the ‘joint_freq’ array row annotation.
- `joint_faf_meta`: Joint filtering allele frequency across the exomes and genomes metadata. An ordered list containing the frequency aggregation group for each element of the ‘joint_faf’ array row annotation. joint_faf_index_dict: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (‘joint_faf’) row annotation.
- `joint_faf_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (‘joint_faf’) row annotation.
- `joint_sample_count`: Number of gnomAD exomes + number of gnomAD genomes.

Row fields

> 

Reviewer: Also for Row fields, main change seems to be "region_flags" and "histograms".
```
Row fields:
    'locus': locus<GRCh38> 
    'alleles': array<str> 
    'region_flags': struct {
        fail_interval_qc: bool, 
        outside_ukb_capture_region: bool, 
        outside_broad_capture_region: bool, 
        outside_ukb_calling_region: bool, 
        outside_broad_calling_region: bool, 
        not_called_in_exomes: bool, 
        not_called_in_genomes: bool
    } 
    'genomes': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32, 
            gen_anc: str
        }, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }
    } 
    'exomes': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int64, 
            gen_anc: str
        }, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }
    } 
    'joint': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str, 
            joint_fafmax_data_type: str
        }, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32, 
            gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }, 
        metric_data_type: str
    } 
    'freq_comparison_stats': struct {
        contingency_table_test: array<struct {
            odds_ratio: float64, 
            p_value: float64
        }>, 
        cochran_mantel_haenszel_test: struct {
            odds_ratio: float64, 
            p_value: float64
        }
    } 
```

Author: the exomes frequencies are also no longer stratified by subset (the exome grpmax field was stratified in a previous schema). just confirming this was intentional @jkgoodrich

Reviewer: When I looked at the exomes release HT, `grpmax` and `fafmax` are still nested struct of `gnomad` and `non_ukb`, I think like last time, we only used these two from `gnomad` to get joint FAF. 
I don't see `freq` was stratified by subsets, in v4.0 release HT and help page.

Author: sounds good, it makes sense to me that you'd use exome-wide frequencies to create the joint freqs.

also, `freq` doesn't need to be stratified because the subset information is in the `freq_meta` stratification (e.g, `{'downsampling': '200000', 'gen_anc': 'global', 'group': 'adj', 'subset': 'non_ukb'}`)

Author: I'm not 100% sure now; that might have been me checking the schema of the v4.0 exomes release HT itself. I vote we resolve this thread

    - `faf95`: Combined exomes and genomes filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Combined exomes and genomes filtering allele frequency (using Poisson 99% CI).
- `joint_grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency across both exomes and genomes. Excludes Amish (ami), Ashkenazi Jewish (asj), European Finnish (fin), and "Remaining individuals" (remaining) groups.
    - `AC`: Alternate allele count in the group with the maximum allele frequency.
    - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
    - `AN`: Total number of alleles in the group with the maximum allele frequency.
    - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
    - `gen_anc`: Genetic ancestry group with maximum allele frequency.

> - `contingency_table_test`
> - `cochran_mantel_haenszel_test`

Reviewer: Needs description for these two. How about: 
```suggestion
- `contingency_table_test`: Hail's `contingency_table_test` on the REF and ALT alleles counts between exomes and genomes. 
- `cochran_mantel_haenszel_test`: Cochran–Mantel–Haenszel test on the REF and ALT alleles counts between exomes and genomes using genetic ancestry group as the stratification.
```

Author: works for me -- I was hoping for @jkgoodrich's help with these (so will leave this open for now)

  - `per_platform`: Whether these thresholds were stratified per platform.
  - `all_platforms`: Whether these thresholds were applied uniformly across platforms, as long as a platform had a sample size above `min_platform_size`.
  - `high_qual_cutoffs`: Interval QC thresholds per chromosomal region:
    - `autosome_par`: Contains annotation used to filter high coverage intervals on autosomes and in pseudoautosomal regions and threshold for annotation. Interval was considered high quality only if this annotation was over the specified threshold value.
    - `x_non_par`: Contains annotation used to filter high coverage intervals in non-pseudoautosomal regions of chromosome X and threshold for annotation. Interval was considered high quality only if this annotation was over the specified threshold value.
    - `y_non_par`: Contains annotation used to filter high coverage intervals in non-pseudoautosomal regions of chromosome Y and threshold for annotation. Interval was considered high quality only if this annotation was over the specified threshold value.

>   - `min_platform_size`: Sample size required for a platform to be considered.

Reviewer: I noticed it's like this in release HT: 
```commandline
        all_platforms: bool, 
        high_qual_cutoffs: dict<str, array<tuple (
            str, 
            str, 
            float64
        )>>, 
        min_platform_size: int32
```
Under high_qual_cutoffs, there isn't any field name.

Author: do you mean that there aren't direct names under `high_qual_cutoffs`? that is true, but if you eval that dict:
```
>>> ht = hl.read_table('gs://gnomad/release/4.1/ht/exomes/gnomad.exomes.v4.1.sites.ht')
>>> hl.eval(ht.interval_qc_parameters)
Struct(per_platform=False, all_platforms=True, high_qual_cutoffs={'autosome_par': [('fraction_over_20x', 'all', 0.85)], 'x_non_par': [('fraction_over_20x', 'XX', 0.85), ('fraction_over_10x', 'XY', 0.85)], 'y_non_par': [('fraction_over_10x', 'XY', 0.85)]}, min_platform_size=100)
```
so I added this description here to make this field easier to understand

  - `freq_meta_sample_count`: A sample count per sample grouping defined in the joint ‘joint.freq_meta’ global annotation.
  - `age_distribution`: Callset-wide age histogram calculated on the combined gnomAD exomes and genomes.
    - `bin_edges`: Bin edges for the age histogram.
    - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
    - `n_smaller`: Count of age values falling below lowest histogram bin edge.
    - `n_larger`: Count of age values falling above highest histogram bin edge.

Row fields

- `locus`: Variant locus. Contains contig and position information.
- `alleles`: Variant alleles.
- `genomes`: Struct of allele frequency information from the gnomAD genomes.
  - `freq`: Array of allele frequency information (AC, AN, AF, homozygote count) for each frequency aggregation group in the gnomAD genomes.
    - `AC`: Alternate allele count in release.
    - `AF`: Alternate allele frequency, (AC/AN), in release.
    - `AN`: Total number of alleles in release.
    - `homozygote_count`: Count of homozygous alternate individuals in release.

>   - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD genomes. Excludes Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.
>     - `AC`: Alternate allele count in the group with the maximum allele frequency.
>     - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
>     - `AN`: Total number of alleles in the group with the maximum allele frequency.
>     - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
>     - `gen_anc`: Genetic ancestry group with maximum allele frequency
>   - `faf`: Filtering allele frequency in the gnomAD genomes.
>     - `faf95`: Filtering allele frequency (using Poisson 95% CI).
>     - `faf99`: Filtering allele frequency (using Poisson 99% CI).

Reviewer: ```suggestion
  - `faf`: Filtering allele frequency in the gnomAD genomes.
    - `faf95`: Filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Filtering allele frequency (using Poisson 99% CI).
  - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD genomes. Excludes Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.
    - `AC`: Alternate allele count in the group with the maximum allele frequency.
    - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
    - `AN`: Total number of alleles in the group with the maximum allele frequency.
    - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
    - `gen_anc`: Genetic ancestry group with maximum allele frequency.
  - `fafmax`: Information about the genetic ancestry group with the maximum filtering allele frequency.
    - `faf95_max`: Maximum filtering allele frequency (using Poisson 95% CI).
    - `faf95_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (95% CI).
    - `faf99_max`: Maximum filtering allele frequency (using Poisson 99% CI).
    - `faf99_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (99% CI).
```

      - `n_smaller`: Count of AB values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above the highest histogram bin edge, calculated on high quality genotypes.
  - `raw_qual_hists`: Genotype quality metric histograms for all genotypes as opposed to high quality genotypes.
    - `gq_hist_all`: Histogram for GQ calculated on all genotypes.
      - `bin_edges`: Bin edges for the GQ histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
      - `bin_freq`: Bin frequencies for the GQ histogram calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of GQ values falling below lowest histogram bin edge, for GQ calculated on all genotypes.
      - `n_larger`: Count of GQ values falling above highest histogram bin edge, for GQ calculated on all genotypes.
    - dp_hist_all: Histogram for DP calculated on all genotypes.

>       - bin_edges: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100

Reviewer: ```suggestion
    - `dp_hist_all`: Histogram for DP calculated on all genotypes.
      - `bin_edges`: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
```

      - `bin_freq`: Bin frequencies for the histogram of DP in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
    - `ab_hist_alt`: Histogram for AB in heterozygous individuals calculated on all genotypes.
      - `bin_edges`: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
      - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.

>     - `age_hists`: Histograms containing age information for release samples.

Reviewer: ```suggestion
  - `age_hists`: Histograms containing age information for release samples.
```

      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
    - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
- `exomes`: Struct of allele frequency information from the gnomAD exomes.
  - `freq`: Array of allele frequency information (AC, AN, AF, homozygote count) for each frequency aggregation group in the gnomAD exomes.
    - `AC`: Alternate allele count in release.
    - `AF`: Alternate allele frequency, (AC/AN), in release.
    - `AN`: Total number of alleles in release.
    - `homozygote_count`: Count of homozygous alternate individuals in release.

>   - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD exomes. Excludes Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.
>     - `AC`: Alternate allele count in the group with the maximum allele frequency.
>     - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
>     - `AN`: Total number of alleles in the group with the maximum allele frequency.
>     - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
>     - `gen_anc`: Genetic ancestry group with maximum allele frequency
>   - `faf`: Filtering allele frequency in the gnomAD exomes.
>     - `faf95`: Filtering allele frequency (using Poisson 95% CI).
>     - `faf99`: Filtering allele frequency (using Poisson 99% CI).

Reviewer: Better to keep as it is ordered in the joint HT: 
```suggestion
  - `faf`: Filtering allele frequency in the gnomAD genomes.
    - `faf95`: Filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Filtering allele frequency (using Poisson 99% CI).
  - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD genomes. Excludes Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.
    - `AC`: Alternate allele count in the group with the maximum allele frequency.
    - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
    - `AN`: Total number of alleles in the group with the maximum allele frequency.
    - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
    - `gen_anc`: Genetic ancestry group with maximum allele frequency.
  - `fafmax`: Information about the genetic ancestry group with the maximum filtering allele frequency.
    - `faf95_max`: Maximum filtering allele frequency (using Poisson 95% CI).
    - `faf95_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (95% CI).
    - `faf99_max`: Maximum filtering allele frequency (using Poisson 99% CI).
    - `faf99_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (99% CI).
```

Author: I started adding changes locally because the mix of committing here and local suggestions was making the formatting wonky -- let me know if you see anything weird slip through

      - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on high quality genotypes. The number of records found in each bin.
      - `n_smaller`: Count of AB values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above the highest histogram bin edge, calculated on high quality genotypes.
  - `raw_qual_hists`: Genotype quality metric histograms for all genotypes as opposed to high quality genotypes.
    - `gq_hist_all`: Histogram for GQ calculated on all genotypes.
      - `bin_edges`: Bin edges for the GQ histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
      - `bin_freq`: Bin frequencies for the GQ histogram calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of GQ values falling below lowest histogram bin edge, for GQ calculated on all genotypes.
      - `n_larger`: Count of GQ values falling above highest histogram bin edge, for GQ calculated on all genotypes.

>     - dp_hist_all: Histogram for DP calculated on all genotypes.
>       - bin_edges: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100
>       - `bin_freq`: Bin frequencies for the DP histogram calculated on all genotypes. The number of records found in each bin.

Reviewer: ```suggestion
    - `dp_hist_all`: Histogram for DP calculated on all genotypes.
      - `bin_edges`: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
```

      - `n_smaller`: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
    - `ab_hist_alt`: Histogram for AB in heterozygous individuals calculated on all genotypes.
      - `bin_edges`: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
      - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
    - `age_hists`: Histograms containing age information for release samples.

>     - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.

Reviewer: ```suggestion
  - `age_hists`: Histograms containing age information for release samples.
```

    - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
- `joint`: Struct of combined (joint) exomes and genomes allele frequency information.
  - `freq`: Array of allele frequency information (AC, AN, AF, homozygote count) across the combined (joint) gnomAD exomes and genomes and for each genetic ancestry group.
    - AC: Combined (exomes + genomes) alternate allele count in release.
    - AF: Combined (exomes + genomes) alternate allele frequency, (AC/AN), in release.
    - AN: Total number of alleles across exomes and genomes in release.
    - homozygote_count: Count of homozygous alternate individuals across exomes and genomes in release.

>   - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency across both exomes and genomes. Excludes Amish (ami), Ashkenazi Jewish (asj), European Finnish (fin), and "Remaining individuals" (remaining) groups.
>     - `AC`: Alternate allele count in the group with the maximum allele frequency.
>     - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
>     - `AN`: Total number of alleles in the group with the maximum allele frequency.
>     - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
>     - `gen_anc`: Genetic ancestry group with maximum allele frequency.
>   - `faf`: Array of combined exomes and genomes filtering allele frequency information (AC, AN, AF, homozygote count). Note that the values in array will correspond to the joint or combined value if the variant had a defined filtering allele frequency in both data types, otherwise this array will contain filtering allele frequencies only for the data type associated with the Hail Table (in this case, exomes).
>     - `faf95`: Combined exomes and genomes filtering allele frequency (using Poisson 95% CI).
>     - `faf99`: Combined exomes and genomes filtering allele frequency (using Poisson 99% CI).

Reviewer: ```suggestion
  - `faf`: Array of combined exomes and genomes filtering allele frequency information (AC, AN, AF, homozygote count). Note that the values in array will correspond to the joint or combined value if the variant had a defined filtering allele frequency in both data types, otherwise this array will contain filtering allele frequencies only for the data type associated with the Hail Table (in this case, exomes).
    - `faf95`: Combined exomes and genomes filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Combined exomes and genomes filtering allele frequency (using Poisson 99% CI).
  - `fafmax`: Information about the genetic ancestry group with the maximum filtering allele frequency.
    - `faf95_max`: Maximum filtering allele frequency (using Poisson 95% CI).
    - `faf95_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (95% CI).
    - `faf99_max`: Maximum filtering allele frequency (using Poisson 99% CI).
    - `faf99_max_gen_anc`: Genetic ancestry group with the maximum filtering allele frequency (99% CI).
  - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency across both exomes and genomes. Excludes Amish (ami), Ashkenazi Jewish (asj), European Finnish (fin), and "Remaining individuals" (remaining) groups.
    - `AC`: Alternate allele count in the group with the maximum allele frequency.
    - `AF`: Maximum alternate allele frequency, (AC/AN), across groups in gnomAD.
    - `AN`: Total number of alleles in the group with the maximum allele frequency.
    - `homozygote_count`: Count of homozygous individuals in the group with the maximum allele frequency.
    - `gen_anc`: Genetic ancestry group with maximum allele frequency.
```

      - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on high quality genotypes. The number of records found in each bin.
      - `n_smaller`: Count of AB values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above the highest histogram bin edge, calculated on high quality genotypes.
  - `raw_qual_hists`: Genotype quality metric histograms for all genotypes as opposed to high quality genotypes.
    - `gq_hist_all`: Histogram for GQ calculated on all genotypes.
      - `bin_edges`: Bin edges for the GQ histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
      - `bin_freq`: Bin frequencies for the GQ histogram calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of GQ values falling below lowest histogram bin edge, for GQ calculated on all genotypes.
      - `n_larger`: Count of GQ values falling above highest histogram bin edge, for GQ calculated on all genotypes.

>     - dp_hist_all: Histogram for DP calculated on all genotypes.
>       - bin_edges: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100

Reviewer: ```suggestion
    - `dp_hist_all`: Histogram for DP calculated on all genotypes.
      - `bin_edges`: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
```

      - `bin_freq`: Bin frequencies for the histogram of DP in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
    - `ab_hist_alt`: Histogram for AB in heterozygous individuals calculated on all genotypes.
      - `bin_edges`: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
      - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
      - `n_smaller`: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
      - `n_larger`: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.

>     - `age_hists`: Histograms containing age information for release samples.

Reviewer: ```suggestion
  - `age_hists`: Histograms containing age information for release samples.
```

    - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
    - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.

>     - `metric_data_type`: Data type associated with metric. One of 'exomes', 'genomes', or 'joint'.
> - `freq_comparison_stats`

Reviewer: ```suggestion
  - `metric_data_type`: Data type associated with metric. One of 'exomes', 'genomes', or 'joint'.

```

> - `freq_comparison_stats`

Reviewer: Tag @jkgoodrich to add descriptions here.

        - `n_smaller`: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
        - `n_larger`: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
      - `ab_hist_alt`: Histogram for AB in heterozygous individuals calculated on all genotypes.
        - `bin_edges`: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
        - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
        - `n_smaller`: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
        - `n_larger`: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.

>       - `age_hists`: Histograms containing age information for release samples.
>         - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
>           - `bin_edges`: Bin edges for the age histogram.
>           - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>           - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>           - `n_larger`: Count of age values falling above highest histogram bin edge.
>         - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
>           - `bin_edges`: Bin edges for the age histogram.
>           - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>           - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>           - `n_larger`: Count of age values falling above highest histogram bin edge.

Reviewer: ```suggestion
    - `age_hists`: Histograms containing age information for release samples.
      - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
        - `bin_edges`: Bin edges for the age histogram.
        - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
        - `n_smaller`: Count of age values falling below lowest histogram bin edge.
        - `n_larger`: Count of age values falling above highest histogram bin edge.
      - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
        - `bin_edges`: Bin edges for the age histogram.
        - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
        - `n_smaller`: Count of age values falling below lowest histogram bin edge.
        - `n_larger`: Count of age values falling above highest histogram bin edge.
```

- `joint`: Struct of combined (joint) exomes and genomes allele frequency information.
  - `freq`: Array of allele frequency information (AC, AN, AF, homozygote count) across the combined (joint) gnomAD exomes and genomes and for each genetic ancestry group.

>     - AC: Combined (exomes + genomes) alternate allele count in release.
>     - AF: Combined (exomes + genomes) alternate allele frequency, (AC/AN), in release.
>     - AN: Total number of alleles across exomes and genomes in release.
>     - homozygote_count: Count of homozygous alternate individuals across exomes and genomes in release.

Reviewer: ```suggestion
    - `AC`: Combined (exomes + genomes) alternate allele count in release.
    - `AF`: Combined (exomes + genomes) alternate allele frequency, (AC/AN), in release.
    - `AN`: Total number of alleles across exomes and genomes in release.
    - `homozygote_count`: Count of homozygous alternate individuals across exomes and genomes in release.
```

        - `n_smaller`: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
        - `n_larger`: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
      - `ab_hist_alt`: Histogram for AB in heterozygous individuals calculated on all genotypes.
        - `bin_edges`: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
        - `bin_freq`: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
        - `n_smaller`: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
        - `n_larger`: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.

>       - `age_hists`: Histograms containing age information for release samples.
>         - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
>           - `bin_edges`: Bin edges for the age histogram.
>           - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>           - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>           - `n_larger`: Count of age values falling above highest histogram bin edge.
>         - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
>           - `bin_edges`: Bin edges for the age histogram.
>           - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>           - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>           - `n_larger`: Count of age values falling above highest histogram bin edge.

Reviewer: ```suggestion
  - `age_hists`: Histograms containing age information for release samples.
    - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
    - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
      - `bin_edges`: Bin edges for the age histogram.
      - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
      - `n_smaller`: Count of age values falling below lowest histogram bin edge.
      - `n_larger`: Count of age values falling above highest histogram bin edge.
```

>   - `age_hists`: Histograms containing age information for release samples.
>     - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
>       - `bin_edges`: Bin edges for the age histogram.
>       - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>       - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>       - `n_larger`: Count of age values falling above highest histogram bin edge.
>     - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
>       - `bin_edges`: Bin edges for the age histogram.
>       - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
>       - `n_smaller`: Count of age values falling below lowest histogram bin edge.
>       - `n_larger`: Count of age values falling above highest histogram bin edge.
>     - `metric_data_type`: Data type associated with metric. One of 'exomes', 'genomes', or 'joint'.

Reviewer: ```suggestion
    - `age_hists`: Histograms containing age information for release samples.
      - `age_hist_het`: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
        - `bin_edges`: Bin edges for the age histogram.
        - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
        - `n_smaller`: Count of age values falling below lowest histogram bin edge.
        - `n_larger`: Count of age values falling above highest histogram bin edge.
      - `age_hist_hom`: Histogram for age in all homozygous release samples calculated on high quality genotypes. If variant is in the pseudoautosomal regions of chrX or chrY, this histogram also includes age counts of hemizygous samples.
        - `bin_edges`: Bin edges for the age histogram.
        - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
        - `n_smaller`: Count of age values falling below lowest histogram bin edge.
        - `n_larger`: Count of age values falling above highest histogram bin edge.
  - `metric_data_type`: Data type associated with metric. One of 'exomes', 'genomes', or 'joint'.
```
Sorry, my bad.

Author: ah thanks for catching!

- `freq_comparison_stats`
  - `contingency_table_test`
    - `odds_ratio`
    - `p_value`
  - `cochran_mantel_haenszel_test`
    - `odds_ratio`

>     - `p_value`

Author: how about just chisq? that's what we used in RMC

  - `vep_config`: VEP configuration to run VEP version with [Hail](https://hail.is/docs/0.2/methods/genetics.html#hail.methods.vep). File created using command within VEP init shell script in https://github.com/broadinstitute/gnomad_methods/tree/main.
  - `gencode_version`: GENCODE version used in VEP.
  - `mane_select_version`: MANE select version used in VEP.

> - `frequency_README`: Explanation of how to use the '`freq_index_dict`' global annotation to extract frequencies from the '`freq`' row annotation.

Reviewer: What's the difference between '`freq`' and `freq`? I saw from L335, you only changed to `genomes.freq`.

Author: Julia added these, but my thinking was that we add quotes here ('`freq`' ) because we're referring both to the name of the annotation and the actual annotation. to refer to `genomes.freq` in the same way, I think we'd want to do
> '`freq'` annotation within the '`genomes`' struct

        - `bin_edges`: Bin edges for the age histogram.
        - `bin_freq`: Bin frequencies for the age histogram. This is the number of records found in each bin.
        - `n_smaller`: Count of age values falling below lowest histogram bin edge.
        - `n_larger`: Count of age values falling above highest histogram bin edge.
- `genomes`: Struct of allele frequency information from the gnomAD genomes.
  - `freq`: Array of allele frequency information (AC, AN, AF, homozygote count) for each frequency aggregation group in the gnomAD genomes.
    - `AC`: Alternate allele count in release.
    - `AF`: Alternate allele frequency, (AC/AN), in release.
    - `AN`: Total number of alleles in release.
    - `homozygote_count`: Count of homozygous alternate individuals in release.
  - `faf`: Filtering allele frequency in the gnomAD genomes.
    - `faf95`: Filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Filtering allele frequency (using Poisson 99% CI).

>   - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD genomes. Excludes Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.

Author: what about AMI for genomes?

Reviewer: Oh, I didn't pay attention to the "AMI", let me double check. Then `mid` wasn't removed from exomes either, right? https://github.com/broadinstitute/gnomad_methods/blob/fde54e45017809c96c77a4da0381c28c3e41ad64/gnomad/resources/grch38/gnomad.py#L229

Author: ah, that's right -- this text would have been from v4.0 where we did remove mid

  - `faf`: Filtering allele frequency in the gnomAD genomes.
    - `faf95`: Filtering allele frequency (using Poisson 95% CI).
    - `faf99`: Filtering allele frequency (using Poisson 99% CI).

>   - `grpmax`: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked genetic ancestry group with maximum allele frequency in the gnomAD genomes. Excludes Amish (`ami`), Ashkenazi Jewish (`asj`), European Finnish (`fin`), Middle Eastern (`mid`), and "Remaining individuals" (`remaining`) groups.

Author: did the genomes not get updated to include mid?

Reviewer: I think `mid` is still too small in terms of samples in genomes.
